docs/dapp/sapphire/quickstart.md

```diff
@@ -0,0 +1,395 @@
+         new HDWalletProvider([process.env.PRIVATE_KEY], "https://testnet.emerald.oasis.dev"),
+       network_id: 0xa515,
+     },
++    sapphire: {
++      provider: () =>
++        new HDWalletProvider([process.env.PRIVATE_KEY], "https://testnet.sapphire.oasis.dev"),
++      network_id: 0x5aff,
++    },
+   },
+
+   // Set default mocha options here, use special reporters etc.
+```
+
+#### Port to Sapphire
+
+Here's where things start to get interesting.
+We're going to add confidentiality to this starter project in exactly two lines
+of code.
+
+You'll need to grab the Sapphire compatibility library
+([@oasisprotocol/sapphire-paratime]), so make that happen by issuing
+
+```sh
+pnpm add @oasisprotocol/sapphire-paratime # npm also works
+```
+
+So far so good.
+Next, import it by adding this line to the top of `truffle-config.js`:
+
+```typescript
+const sapphire = require('@oasisprotocol/sapphire-paratime');
+```
+
+That's the first line of code.
+Here's the second:
+
+```diff
+diff --git a/truffle-config.js b/truffle-config.js
+index 0cd9d36..7db7cf8 100644
+--- a/truffle-config.js
++++ b/truffle-config.js
+@@ -60,7 +60,7 @@ module.exports = {
+     },
+     sapphire: {
+       provider: () =>
+-        new HDWalletProvider([process.env.PRIVATE_KEY], "https://testnet.sapphire.oasis.dev"),
++        sapphire.wrap(new HDWalletProvider([process.env.PRIVATE_KEY], "https://testnet.sapphire.oasis.dev")),
+       network_id: 0x5aff,
+     },
+   },
+```
+
+This `wrap` function takes any kind of provider or signer you've got and turn
+it into one that works with Sapphire and confidentiality.
+For the most part, wrapping your signer/provider is the most you'll need to do
+to get your dapp rollin' on Sapphire, but that's not a complete story since an
+unmodified contract may leak state through normal operation.
+
+And now for the moment we've all been waiting for:
+
+```
+> PRIVATE_KEY="0x..." truffle migrate --network sapphire
+
+Sent some coins in 0x6dc6774addf4c5c68a9b2c6b5e5634263e734d321f84012ab1b4cbe237fbe7c2.
+A Transfer(0x56e5F834F88F9f7631E9d6a43254e173478cE06a, 0x56e5F834F88F9f7631E9d6a43254e173478cE06a, 42) was emitted.
+The balance storage slot contains 0x0.
+The contract now has balance: 42.
+```
+
+So basically nothing changed, which is pretty much what we're going for.
+But take a look at that second to last line where it says what's in the storage
+slot. Before, it said `0x2a`, but now it says `0x0`.
+
+Clearly the slot does contain data or else the contract balance couldn't have
+been returned.  
+What's happened here is that the Web3 gateway does not have the key used to
+decrypt the storage slot, so a default value is returned.
+
+Indeed, the gateway does not even have the key needed to decrypt the _key_ in
+the MKVS; it can tell _that_ a storage slot was written, but not which one
+(although it can make a very good guess by reading the contract code).
+So you can see that confidentiality is in effect, but it's not something
```

Comment:
This sentence is not really related to the previous one about the key of the storage slot.

```suggestion

Finally, you can see that confidentiality is in effect out of the box, but it's not something
```